lib/jekyll-admin/server/collection.rb

```diff
@@ -7,32 +7,34 @@ class Server < Sinatra::Base
 
       get "/:collection_id" do
         ensure_collection
-        json collection.to_api(:include_documents => true)
+        json collection.to_api
       end
 
-      get "/:collection_id/*" do
+      get "/:collection_id/*?/?:path.:ext" do
         ensure_document
         json document.to_api(:include_content => true)
       end
 
-      put "/:collection_id/*" do
-        ensure_collection
+      get "/:collection_id/entries/?*" do
+        ensure_directory
+        json entries.map(&:to_api)
+      end
 
-        # Rename page
-        if request_payload["path"]
-          request_payload["path"].gsub!(%r!\A_#{collection.label}/!, "")
-          if request_payload["path"] != params["splat"].first
-            delete_file document_path
-            params["splat"] = [request_payload["path"]]
-          end
+      put "/:collection_id/*?/?:path.:ext" do
+        ensure_collection
+        write_path = document_path
+        if request_payload["path"] && request_payload["path"] != relative_document_path
+          delete_file document_path
+          write_path = request_path
         end
 
-        write_file(document_path, document_body)
-        ensure_document
-        json document.to_api(:include_content => true)
+        write_file(write_path, document_body)
+        updated_document = collection.docs.find { |d| d.path == write_path }
+        render_404 if updated_document.nil?
```

Comment:
I know this doesn't look good but `ensure_document` uses `document` which relies on `document_path` for comparison. We need `write_path` here to compare with.